x-pack/plugins/dashboard_enhanced/common/drilldowns/dashboard_drilldown/dashboard_drilldown_persistable_state.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectReference } from '../../../../../../src/core/types';
+import { PersistableStateDefinition } from '../../../../../../src/plugins/kibana_utils/common';
+import { SerializedAction, SerializedEvent } from '../../../../ui_actions_enhanced/common/types';
+import { DrilldownConfig } from './types';
+
+export type DashboardDrilldownPersistableStateDefinition = PersistableStateDefinition<
+  SerializedEvent
+>;
+
+const generateRefName = (state: SerializedEvent, id: string) =>
+  `drilldown:${id}:${state.eventId}:dashboardId`;
```

Comment:
Do we have a naming convention for reference names?

Reply:
not yet, i agree it would be useful to have one